ios/PaymentsRequest.swift

```diff
@@ -19,6 +19,20 @@ internal struct PaymentsData {
     static var merchantAccount : String = ""
     static var additionalData : [String : Any] = ["allow3DS2": true,"executeThreeD":true]
     static var regionId : Int = 1
+    /// Initial cost to unlock
+    static var baseRate: Int?
+    /// Cost per some unit of time
+    static var effectiveRate: Int?
+    /// Localized string to present before line items (e.g. Cost per ride varies)
+    static var localizedNativePayTitle: String = "Cost per ride varies"
```

Comment:
This must get updated with a true localized string somewhere, right? Seems funny to name these variables `localized...` just to hard-code an english string (I mean, I guess it has to initialize to _something_)

Reply:
> The `AdyenPayment.swift` file could use some love at some point 🙂

Agreed, it feels like it was a StackOverflow copy/paste job. The code in `PaymentServices` is similar, which that should definitely be refactored at some point if it's intent is to outlive this RN shim…

> This must get updated with a true localized string somewhere, right? Seems funny to name these variables `localized...` just to hard-code an english string (I mean, I guess it has to initialize to _something_)

Yes, RN _should_ be setting these values and that is the basis of my next PR related to this in `spin-mobile`. I definitely should have init'd them as localized for sake of intent to the reader even if the localizations will never need to come from within this bundle so that's been updated 🙃

Reply:
We started reimplementing this for our native add card flow, which is part of PaymentServices. My intent is for the app clip to eventually ditch the vendored adyen stuff and use something we can actually maintain.

For the RN bridge code, not sure how much we want to invest in it.

Reply:
Exactly how I feel.